securedrop_client/gui/widgets.py

```diff
@@ -1898,6 +1925,46 @@ def set_failed_to_decrypt_styles(self) -> None:
         self.color_bar.setStyleSheet(self.STATUS_BAR_CSS)
         self.sender_icon.set_failed_to_decrypt_styles()
 
+    def update_seen_by_list(self, usernames: Dict[str, User]) -> None:
+        # Update the dictionary for the new usernames to be shown in the tooltip.
+        self.seen_by.update(usernames)
+
+        # Re-arrange for the authenticated user's username to be shown at the end of the
+        # username list shown in the tooltip.
+        if self.authenticated_user and self.authenticated_user.username in self.seen_by:
+            del self.seen_by[self.authenticated_user.username]
+            self.seen_by[self.authenticated_user.username] = self.authenticated_user
+
+        self.check_mark.setToolTip(",\n".join(str(username) for username in self.seen_by.keys()))
+
+    def eventFilter(self, obj: QObject, event: QEvent) -> None:
+        t = event.type()
+        if t == QEvent.HoverEnter:
+            self.check_mark.setIcon(load_icon("checkmark_hover.svg"))
+        elif t == QEvent.HoverLeave:
+            self.check_mark.setIcon(load_icon("checkmark.svg"))
+
+        return QObject.event(obj, event)
+
+
+class CheckMark(QPushButton):
+    """
+    Represents the seen by checkmark for each bubble.
+    """
+
+    CHECK_MARK_CSS = load_css("checker_tooltip.css")
+    CLICKABLE_SPACE = 60
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.setObjectName("Checker")
+        self.setStyleSheet(self.CHECK_MARK_CSS)
+        layout = QHBoxLayout()
+        self.setIcon(load_icon("checkmark.svg"))
+        self.setIconSize(QSize(16, 9))
```

Comment:
I tried keeping the size dimensions as close to the Figma file reference as possible, thus the difference.

Reply:
Subtle dilemma... I would personally go for 16x10 in that case, with the understanding that the SVG files were created from the Figma file, and if they should be 16x9... well, then the SVG files should be updated.

What concerns me a little about resizing the SVGs by a single pixel is that the aspect ratio is modified (so the image is potentially distorted). Whether or not that actually happens depends on Qt rendering strategies and is out of our control.

Additionally, and that's a way more nit-picky concern: it is very unlikely that resizing from 16x10 to 16x9 will result in sharp rendering. Because hi-DPI monitor are becoming more common, and those tend to operate at a 2x scale, it is usually prudent to both create the SVGs on even grids (that is placing lines that are 2px wide, only on even grid columns) and scaling them on even number of pixels (e.g. duplicating or halving the size). If the files were created as 16x10, then I think it is unlikely any of those properties will be preserved if the file is resized to 16x9.

**Summary**: if the result is not visually shocking, I'd go for 16x10 consistently. If it is shocking... I'd have a chat with @ninavizz to get SVG that are natively sized (whether that's 16x10 or 16x9 or any other adequate size).